### DISCUSSION TITLE
LoadTile: easier way to delay or retry loading tiles or to catch an error

### DISCUSSION THREAD

**Discussion by bufemc** [Leaflet/Leaflet#9510] [Ideas]
I am using HERE.com maps with the latest version of Leaflet. Currently, some tiles load initially, but it seems the service gets overwhelmed (or DDOS-ed?), and some tiles then fail to load, resulting in a 429 Too Many Requests error. My idea is to implement an (optional! default would be 0) minimal delay when loading tiles, e.g. in the microsecond range. Or to retry loading after a certain number of milliseconds, up to x times. Additionally, I would like to handle/detect the case where a tile fails to load somehow easier, e.g. by catching the HTTP status code, so you can workaround that case in some way.

**Comment by IvanSanchez** [comment]
On one hand, see https://gitlab.com/IvanSanchez/Leaflet.TileLayer.HERE (by yours truly); some vendors don't like when you don't use their documented APIs.

On the other hand, you might wanna go into `L.TileLayer.createTile()` and delay the execution of this line of code (by means of replacing the method, or extending the class into something like `L.TileLayer.Delay`; see the tutorials about that):

https://github.com/Leaflet/Leaflet/blob/142f94a9ba5757f7e7180ffa6cbed2b3a9bc73c9/src/layer/tile/TileLayer.js#L167

e.g. 
```
setTimeout(()=>{ tile.src = this.getTileUrl(coords); }, 500)
```
